### PR TITLE
Run tests as ECMAScript modules

### LINKED ISSUE #113: Can't use after installing via npm (without webpack or similar)
The project can't be used in the browser after installing it via npm because the files reference

import _Object$keys from "@babel/runtime-corejs3/core-js/object/keys";
import _sliceInstanceProperty from "@babel/runtime-corejs3/core-js/instance/slice";
import _Array$from from "@babel/runtime-corejs3/core-js/array/from";
import _Symbol from "@babel/runtime-corejs3/core-js/symbol";
import _getIteratorMethod from "@babel/runtime-corejs3/core-js/get-iterator-method";
import _getIterator from "@babel/runtime-corejs3/core-js/get-iterator";

causing

Uncaught TypeError: Error resolving module specifier “@babel/runtime-corejs3/core-js/object/keys”. Relative module specifiers must start with “./”, “../” or “/”.

Solution:

Publish truly pre-compiled artifacts (e.g. annotator.min.js) on npm that can be used in the browser without webpack.

### PR DESCRIPTION
**Pull request by tilgovi**
Upgrade Mocha to v9, replace chai with the built-in assert module, replace optimal-select with @medv/finder, replace nyc with c8, and configure the test suite to run without transforming modules to CommonJS.

Close #113.

### CODE REVIEW

File: packages/dom/src/css.ts [hunk 1]
```diff
@@ -20,10 +20,10 @@
  * under the License.
  */
 
-import optimalSelect from 'optimal-select';
+import { finder } from '@medv/finder';
 import type { CssSelector, Matcher } from '@apache-annotator/selector';
-import { ownerDocument } from './owner-document';
-import { toRange } from './to-range';
+import { ownerDocument } from './owner-document.js';
+import { toRange } from './to-range.js';
```
Comment (Treora) [inline packages/dom/src/css.ts:26]: I like the explicitness of adding extensions, but it seems to break now when running `yarn run web:build` or `web:server`, with errors like:

> ERROR in ../packages/dom/src/index.ts
> Module not found: Error: Can't resolve './css.js' in '/vagrant/annotator/incubator-annotator/packages/dom/src'

Renaming all the `.js` imports to `.ts` seems to fix this for the `web:…` scripts; my editor (VSCodium) now does not understand the imports anymore though, not sure why. Also the linter now gives errors, not sure that is a step forward or back. Ideas?

**Review by Treora — COMMENTED** [review]
Look mostly good; just the `.js` endings in the imports give me some trouble.

File: packages/dom/src/css.ts (hunk 1 above)
Comment (tilgovi) [reply packages/dom/src/css.ts:26]: This issue should be resolved now.

File: babel.config.js [hunk 2]
```diff
@@ -53,8 +53,7 @@ module.exports = (api) => {
     ...(DEV || TEST
       ? {
           alias: {
-            '^@apache-annotator/([^/]+)$': ([, name]) =>
-              path.join(packagePath, name, '/src/index.ts'),
+            '^@apache-annotator/([^/]+)$': `${packagePath}/\\1/src/index.ts`,
```
Comment (Treora) [inline babel.config.js:56]: This last commit is exactly reversing the fix for Windows in https://github.com/apache/incubator-annotator/pull/105/files

File: babel.config.js [hunk 3]
```diff
@@ -47,14 +50,25 @@ module.exports = (api) => {
...
+            if (
+              currentFile.startsWith(packagePath) &&
+              currentFile.endsWith('.ts') &&
+              sourcePath.startsWith('.') &&
+              sourcePath.endsWith('.js')
+            ) {
+              return sourcePath.replace(/\.js$/, '.ts');
+            }
+            return resolvePath(sourcePath, currentFile, opts);
+          },
```
Comment (Treora) [inline babel.config.js:68]: Unfortunate we’d need a custom resolver for this, but so be it. I wonder why TypeScript does not simply accept an `import '….ts';` statement, and would transpile it to the equivalent `.js` import.

Comment (tilgovi) [reply babel.config.js:68]: I don't recall the arguments, but I have read through the issues about it. It doesn't seem likely. They don't want to get into the business of rewriting imports. Their position is that you should write `.js`. The TypeScript resolver knows how to resolve that to equivalent `.ts` files, and maybe part of the answer is to use `eslint-import-resolver-typescript`. However, it's helpful to have Babel for our final build anyway, and `babel-plugin-module-resolver` _will_ rewrite imports for us, and `eslint-import-resolver-babel-module` can piggyback off that, so this is what I think is the best thing for now.

File: babel.config.js (hunk 2 above)
Comment (tilgovi) [reply babel.config.js]: Good catch. Pushed up another commit to fix that.

**Pull request merged**